core/transaction_pool.go

```diff
@@ -294,7 +294,22 @@ func (self *TxPool) Add(tx *types.Transaction) (err error) {
 	self.mu.Lock()
 	defer self.mu.Unlock()
 
-	err = self.add(tx)
+	err = self.add(tx, false)
+	if err == nil {
+		// check and validate the queueue
+		self.checkQueue()
+	}
+
+	return
+}
+
+// AddLocal queues a locally created transaction in the pool if it is valid.
+// (validity check does not include comparing gas price to miner's minimum)
+func (self *TxPool) AddLocal(tx *types.Transaction) (err error) {
```

Comment:
Maybe consider instead providing a `local bool` flag to `AddTransactions`? I'm not sure we need to separate method calls for it. Similarly, if the old one is `AddTransactions`, a new version should be `AddLocalTransactions` (note, plural) to keep the naming and parametrization consistent.